cpp/include/raft/handle.hpp

```diff
@@ -126,26 +142,42 @@ class handle_t {
     return cusparse_handle_;
   }
 
-  cudaStream_t get_internal_stream(int sid) const { return streams_[sid]; }
-  int get_num_internal_streams() const { return num_streams_; }
+  // legacy compatibility for cuML
+  cudaStream_t get_internal_stream(int sid) const {
+    return streams_.get_stream(sid).value();
+  }
+  // new accessor return rmm::cuda_stream_view
+  rmm::cuda_stream_view get_internal_stream_view(int sid) const {
+    return streams_.get_stream(sid);
+  }
+
+  int get_num_internal_streams() const { return streams_.get_pool_size(); }
   std::vector<cudaStream_t> get_internal_streams() const {
     std::vector<cudaStream_t> int_streams_vec;
-    for (auto s : streams_) {
-      int_streams_vec.push_back(s);
+    for (int i = 0; i < get_num_internal_streams(); i++) {
+      int_streams_vec.push_back(get_internal_stream(i));
     }
     return int_streams_vec;
   }
 
+  handle_t get_handle_from_internal_pool(
+    int stream_id, int n_streams = kNumDefaultWorkerStreams) const {
+    handle_t handle(n_streams);
+    handle = *this;
```

Comment:
If the copy constructor behavior is idiomatic, we can do ```auto handle = *this;``` instead of the above two lines. But the behavior of the copy constructor is non-idiomatic, I assume this is not possible, and I guess this can be quite surprising to many C++ programmers.

Reply:
What I really need is a light copy of some fields but not the whole handle. Since the compiler wanted a copy constructor because of the return type I just leveraged this to address what I needed it but yea I agree it's a bit hacky :)
I can probably just add a canonical copy constructor and a `get_light_copy()` feature.

Reply:
Actually, I did a small redesign I think it is clearer this way and less intrusive on the existing handle design. Let me know how it looks.